### PR TITLE
[prometheus_sd/kubernetes]add new node address types for discover

### PR DESCRIPTION
**Pull request by tariq1890**
Please see here: https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-apiserver/app/options/options.go#L104

### CODE REVIEW

**Comment by tariq1890** [comment]
@brian-Brazil. Requesting your review

File: discovery/kubernetes/node.go [hunk 1]
```diff
@@ -207,9 +209,15 @@ func nodeAddress(node *apiv1.Node) (string, map[apiv1.NodeAddressType][]string,
 		m[a.Type] = append(m[a.Type], a.Address)
 	}
 
+	if addresses, ok := m[apiv1.NodeInternalDNS]; ok {
```
Comment (brian-brazil) [inline discovery/kubernetes/node.go]: I think we should prefer IP to DNS, as that avoids having to do additional DNS lookups.

Comment (tariq1890) [reply discovery/kubernetes/node.go]: Oh yes, that makes sense. Will do.

File: discovery/kubernetes/node.go [hunk 2]
```diff
@@ -207,9 +209,15 @@ func nodeAddress(node *apiv1.Node) (string, map[apiv1.NodeAddressType][]string,
...
+		return addresses[0], m, nil
+	}
 	if addresses, ok := m[apiv1.NodeInternalIP]; ok {
 		return addresses[0], m, nil
 	}
+	if addresses, ok := m[apiv1.NodeExternalDNS]; ok {
```
Comment (brancz) [inline discovery/kubernetes/node.go]: same here

**Comment by tariq1890** [comment]
Review comments addressed.

**Comment by brancz** [comment]
lgtm :+1: leaving last call up to @brian-brazil

**Comment by brian-brazil** [comment]
You've got a stray file here.

**Comment by tariq1890** [comment]
Thanks for catching that @brian-brazil!

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**